packages/image_picker/tizen/src/image_picker_tizen_plugin.cc

```diff
@@ -151,131 +140,155 @@ class ImagePickerTizenPlugin : public flutter::Plugin {
+    RET_IF_ERROR(ret);
+
     ret = app_control_set_mime(handle, mime_type_.c_str());
-    mime_type_ = "";
     RET_IF_ERROR(ret);
 
-    ret = app_control_send_launch_request(handle, PickImageReplyCb, this);
+    ret = app_control_send_launch_request(handle, PickImageReplyCallback, this);
     RET_IF_ERROR(ret);
+#undef RET_IF_ERROR
 
     app_control_destroy(handle);
   }
 
-#ifndef TV_PROFILE
-  static void AppRequestResponseCb(ppm_call_cause_e cause,
-                                   ppm_request_result_e result,
-                                   const char *privilege, void *data) {
-    ImagePickerTizenPlugin *plugin = (ImagePickerTizenPlugin *)data;
-    assert(plugin);
-
-    if (cause == PRIVACY_PRIVILEGE_MANAGER_CALL_CAUSE_ERROR) {
-      LOG_ERROR("app_request_response_cb failed! [%d]", result);
-      plugin->SendResultWithError("Invalid permission");
-      return;
-    }
-
-    plugin->PickContent();
-  }
-#endif
-
-  static void PickImageReplyCb(app_control_h request, app_control_h reply,
-                               app_control_result_e result, void *user_data) {
-    LOG_INFO("PickImageReplyCb called: %d", (int)result);
+  static void PickImageReplyCallback(app_control_h request, app_control_h reply,
+                                     app_control_result_e result,
+                                     void *user_data) {
+    LOG_INFO("PickImageReplyCallback called: %d", (int)result);
 
-    ImagePickerTizenPlugin *plugin = (ImagePickerTizenPlugin *)user_data;
+    auto *plugin = (ImagePickerTizenPlugin *)user_data;
     assert(plugin != nullptr);
     assert(plugin->result_ != nullptr);
 
     if (result != APP_CONTROL_RESULT_SUCCEEDED) {
-      plugin->SendResultWithError("Not Found Images");
+      plugin->SendResultWithError("Operation failed");
       return;
     }
 
-    char **value = nullptr;
+    char **values = nullptr;
     int count = 0;
     int ret = app_control_get_extra_data_array(reply, APP_CONTROL_DATA_SELECTED,
-                                               &value, &count);
+                                               &values, &count);
     if (ret != APP_CONTROL_ERROR_NONE) {
       plugin->SendResultWithError(std::to_string(ret), get_error_message(ret));
       return;
     }
 
-    if (count == 1) {
-      LOG_INFO("image path: %s", value[0]);
-      std::string src_path = value[0];
-      std::string dst_path;
-      if (plugin->image_resize_.Resize(src_path, dst_path)) {
-        plugin->SendResultWithSuccess(dst_path);
-      } else {
-        plugin->SendResultWithSuccess(src_path);
+    if (count == 0) {
+      plugin->SendResultWithError("No file selected");
+    } else if (plugin->multi_image_) {
+      flutter::EncodableList paths;
+      for (int i = 0; i < count; i++) {
+        std::string source_path = values[i];
+        std::string dest_path;
+        if (plugin->image_resize_.Resize(source_path, dest_path)) {
```

Comment:
It is not the scope of this PR, but It seems that `dest_path` is not used  in` ImageResize::Resize`

Reply:
It's used.

https://github.com/flutter-tizen/plugins/blob/9f53df393ddbb1034d46be190e88caf67af6482d/packages/image_picker/tizen/src/image_resize.cc#L192

Reply:
I misunderstood :laughing: